src/main/java/com/spikes2212/command/genericsubsystem/GenericSubsystem.java

```diff
@@ -10,27 +10,42 @@
  *
  * @author Yuval Levy
  */
-public abstract class GenericSubsystem extends SubsystemBase {
+public abstract class GenericSubsystem extends DashboardedSubsystem {
 
     private double currentSpeed = 0;
     protected Supplier<Double> maxSpeed;
     protected Supplier<Double> minSpeed;
 
     /**
-     * Constructs a new instance of GenericSubsystem.
+     * Constructs a new instance of GenericSubsystem. <br>
+     *
+     * This constructor is deprecated. <br>
+     * Please use {@code GenericSubsystem(String namespaceName)}.
      */
+
+    @Deprecated(since = "2022")
     public GenericSubsystem() {
-        this(-1, 1);
+        this("generic subsystem");
```

Comment:
this is actually a little complicated since you can't call this before `this` or `super`. I can see this working if we add a `setName` method to `Namespace` but I don't want to break anything so I'll think about this later.

Reply:
after some digging, i came up with [this](https://carbon.now.sh/?bg=rgba%28171%2C+184%2C+195%2C+1%29&t=seti&wt=none&l=text%2Fx-java&width=680&ds=true&dsyoff=20px&dsblur=68px&wc=true&wa=true&pv=56px&ph=56px&ln=false&fl=1&fm=Hack&fs=14px&lh=133%25&si=false&es=2x&wm=false&code=%2520%2520%2520%2520public%2520static%2520String%2520getSourceClassName%28String%2520defaultName%29%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520StackTraceElement%255B%255D%2520stes%2520%253D%2520new%2520Throwable%28%29.getStackTrace%28%29%253B%250A%2520%2520%2520%2520%2520%2520%2520%2520String%2520str%2520%253D%2520defaultName%253B%250A%2520%2520%2520%2520%2520%2520%2520%2520for%28int%2520i%253Dstes.length-1%253Bi%253E%253D0%253Bi--%29%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520if%2520%28stes%255Bi%255D.toString%28%29.contains%28%2522.%253Cinit%253E%2522%29%29%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520str%2520%253D%2520stes%255Bi%255D.getClassName%28%29%253B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520break%253B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%2520%2520%2520return%2520str%253B%250A%2520%2520%2520%2520%257D):
its kinda ugly, but its working...